service/application/src/test/java/com/solace/maas/ep/event/management/agent/service/SolaceResourceConfigurationToEventConverterTests.java

```diff
@@ -0,0 +1,169 @@
+        assertThat(mappedEvent.getMessagingServiceType()).isEqualTo(resourceConfiguration.getResourceConfigurationType().name());
+
+        List<ConnectionDetailsEvent> connections = mappedEvent.getConnectionDetails();
+        assertThat(connections).hasSize(1);
+        ConnectionDetailsEvent connectionDetails = connections.get(0);
+        assertThat(connectionDetails.getName()).isEqualTo(resourceConfiguration.getName());
+        assertThat(connectionDetails.getUrl()).isEqualTo(connection.getUrl());
+
+        assertThat(connectionDetails.getProperties()).containsExactly(EventProperty.builder().name("sempPageSize").value("100").build(),
+                EventProperty.builder().name("msgVpn").value(connection.getMsgVpn()).build());
+
+        assertThat(connectionDetails.getAuthenticationDetails()).hasSize(1);
+        AuthenticationDetailsEvent authenticationDetailsEvent = connectionDetails.getAuthenticationDetails().get(0);
+        assertThat(authenticationDetailsEvent.getProtocol()).isEqualTo(PROTOCOL_SEMP);
+        assertThat(authenticationDetailsEvent.getProperties())
+                .containsExactlyInAnyOrder(EventProperty.builder().name("type").value(BASIC_AUTHENTICATION).build()
+
+                );
+
+        assertThat(authenticationDetailsEvent.getCredentials()).hasSize(1);
+        CredentialDetailsEvent credentialDetailsEvent = authenticationDetailsEvent.getCredentials().get(0);
+        assertThat(credentialDetailsEvent.getProperties())
+                .containsExactlyInAnyOrder(
+                        EventProperty.builder().name("username").value(cred.getUserName()).build(),
+                        EventProperty.builder().name("password").value(cred.getPassword()).build()
+
+                );
+
+
+    }
+
+    @Nested
+    class BeanValidations {
+        @Test
+        void nullInputFailsValidationCheck() {
+            NullPointerException thrown = Assert.assertThrows(NullPointerException.class,
+                    () -> solaceResourceConfigurationToEventConverter.mapToMessagingServiceEvent(null));
+            assertThat(thrown.getMessage()).isEqualTo("eventBrokerResource cannot be null.");
+        }
+
+
+        @Test
+        void beanValidationFailsIfBrokerResourceConfigurationIsInvalid() {
+            EventBrokerResourceConfiguration invalidResource = buildResourceConfiguration();
+            invalidResource.setId(null);
+            invalidResource.setName(null);
+            invalidResource.setConnections(List.of());
+            invalidResource.setResourceConfigurationType(ResourceConfigurationType.SOLACE);
+
+            validateConstraintViolationException(invalidResource, Set.of("mapToMessagingServiceEvent.eventBrokerResource.name: must not be blank",
+                    "mapToMessagingServiceEvent.eventBrokerResource.id: must not be blank", "mapToMessagingServiceEvent.eventBrokerResource.connections: must" +
+                            " not be empty"));
```

Comment:
💭 The path includes the method name too? Weird. There must be config that removes that part. Do we care though? Probably not.